### PR TITLE
Fixes for trust domain configuration

### PR DESCRIPTION
**Pull request by howardjohn**
We want to ensure we take values.global.trustDomain as default, fallback
to meshConfig.trustDomain, and ensure this is passed through to all
parts of the code.

This fixes the breakage in https://github.com/istio/istio.io/pull/8301

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/cmd/pilot-agent/main.go [hunk 1]
```diff
@@ -127,6 +126,9 @@ var (
 	enableGatewaySDSEnv = env.RegisterBoolVar("ENABLE_INGRESS_GATEWAY_SDS", false,
 		"Enable provisioning gateway secrets. Requires Secret read permission").Get()
 
+	trustDomainEnv = env.RegisterStringVar("TRUST_DOMAIN", "cluster.local",
```
Comment (howardjohn) [inline pilot/cmd/pilot-agent/main.go:129]: This was removed in #27418 which is backwards incompatible, we have users relying on setting this. The flag should be removed instead, as done in this PR, IMO.

File: manifests/charts/gateways/istio-egress/templates/deployment.yaml [hunk 2]
```diff
@@ -121,9 +121,6 @@ spec:
         {{- if .Values.global.sts.servicePort }}
           - --stsPort={{ .Values.global.sts.servicePort }}
         {{- end }}
-        {{- if .Values.global.trustDomain }}
-          - --trust-domain={{ .Values.global.trustDomain }}
-        {{- end }}
```
Comment (hzxuzhonghu) [inline manifests/charts/gateways/istio-egress/templates/deployment.yaml:126]: nit: You can leave it here, #28048 remove it

**Review by costinm — APPROVED** [review]

File: pilot/cmd/pilot-agent/main.go (hunk 1 above)
Comment (howardjohn) [reply pilot/cmd/pilot-agent/main.go:129]: we ignore them

**Review by hzxuzhonghu — APPROVED** [review]

**Review by morvencao — APPROVED** [review]

**Comment by bplotnick** [comment]
I believe this affects 1.7. The docs imply automatic deprecation for the `values` form if the parameter is available in `meshConfig`:

```
Anything that is available in IstioOperatorSpec should be set above rather than using the passthrough. This includes Kubernetes resource settings for components in KubernetesResourcesSpec.
```

Given this, I think it's worth backporting.

**Pull request merged**